src/Templates/ModelTemplate.cshtml

```diff
@@ -7,9 +7,21 @@
 
 @inherits AutoRest.Core.Template<AutoRest.Go.Model.CompositeTypeGo>
 
+@{
+    var depMessage = "This type has been deprecated.";
+    if (!string.IsNullOrWhiteSpace(Model.DeprecationMessage))
+    {
+        depMessage = Model.DeprecationMessage;
+    }
+}
+
```

Comment:
was thinking, for when the future Types are created, do they get the deprecation fields so we can check to deprecate them ?

Reply:
They do but unfortunately there's not private setter so I can't set Deprecated to true upon construction.

Reply:
We can try to set the `DeprecationMessage` and should be enough [https://github.com/Azure/autorest.common/blob/9f6f2ed29dbe5561e4319c4672c41721fdb7ee06/src/Model/IModelType.cs](https://github.com/Azure/autorest.common/blob/9f6f2ed29dbe5561e4319c4672c41721fdb7ee06/src/Model/IModelType.cs)

---
In reply to: [166104232](https://github.com/Azure/autorest.go/pull/90#discussion_r166104232) [](ancestors = 166104232)

Reply:
Fixed for future and page types.